server/datastore/mysql/migrations/tables/20230721161508_QueriesDataMigrator.go

```diff
@@ -0,0 +1,458 @@
+			)
+		}
+	}
+	if err := rows.Err(); err != nil {
+		return fmt.Errorf("error on 'rows' for non-scheduled queries: %s", err)
+	}
+	if err := rows.Close(); err != nil {
+		return fmt.Errorf("error executing 'Close' for non-scheduled queries: %s", err)
+	}
+
+	if len(args) == 0 {
+		return nil
+	}
+
+	insertStmt := `
+		INSERT INTO queries (
+			name,
+			description,
+			query,
+			author_id,
+			saved,
+			observer_can_run,
+			team_id,
+			team_id_char
+		) VALUES %s`
+
+	placeHolders := strings.TrimSuffix(strings.Repeat("( ?, ?, ?, ?, ?, ?, ?, ? ),", nRows), ",")
+	if _, err = tx.Exec(fmt.Sprintf(insertStmt, placeHolders), args...); err != nil {
+		return fmt.Errorf("error executing 'Exec' on non-scheduled queries: %s", err)
+	}
+	return nil
+}
+
+func _20230719152138_clean_up(tx *sql.Tx) error {
+	// Remove query stats
+	if _, err := tx.Exec(`TRUNCATE scheduled_query_stats`); err != nil {
+		return fmt.Errorf("error truncating 'scheduled_query_stats': %s", err)
+	}
+	if _, err := tx.Exec(`DELETE FROM aggregated_stats WHERE type = 'query' OR type = 'scheduled_query'`); err != nil {
+		return fmt.Errorf("error removing aggregated_stats: %s", err)
+	}
+
+	// Delete queries that only belong to 'global' and 'team' packs
+	if _, err := tx.Exec(`
+DELETE
+FROM queries
+WHERE name IN (SELECT query_name
+               FROM (SELECT query_name
+                     FROM scheduled_queries
+                              INNER JOIN packs p on scheduled_queries.pack_id = p.id
+                     WHERE p.pack_type = 'global'
+                     UNION
+                     SELECT query_name
+                     FROM scheduled_queries
+                              INNER JOIN packs p on scheduled_queries.pack_id = p.id
+                     WHERE p.pack_type LIKE 'team-%') r
+               WHERE query_name NOT IN (SELECT query_name
+                                        FROM scheduled_queries
+                                                 INNER JOIN packs p on scheduled_queries.pack_id = p.id
+                                        WHERE p.pack_type IS NULL))	
+	
+	`); err != nil {
+		return fmt.Errorf("error deleting queries that belong only to global / team packs: %s", err)
+	}
+
+	// Remove 'global' and 'team' packs ... relevant rows in the 'scheduled_queries' table should be
+	// deleted because of the on cascade delete on the pack_id FK.
+	if _, err := tx.Exec(`DELETE FROM packs WHERE pack_type = 'global' OR pack_type LIKE 'team-%'`); err != nil {
+		return fmt.Errorf("error deleting packs: %s", err)
+	}
+
+	return nil
+}
+
+func Up_20230721161508(tx *sql.Tx) error {
+	// Migrates 'old' scheduled queries to the 'new' query schema.
+	// Queries can either be:
+	// 	1 - Scheduled, which can either belong to:
+	//		1.1 - The global pack:
+	// 			For each scheduled query create a single global scheduled query named `$query.name - $scheduled.id`.
+	//		1.2 - Team pack:
+	//			Create a new team query with the name of `$query.name - $scheduled.id`.
```

Comment:
It's somewhat unlikely but not impossible that we'll have duplicate names with our new naming scheme (that could conflict with queries that existed before the migration, e.g. if there was a "foo" and a "foo - 2" query, the "foo" one could be migrated to "foo - 2" if the schedule ID was 2, resulting in a conflict). I don't have a good global overview of the changes yet, could that become an issue?

Reply:
Good point - I could add the timestamp of the schedule at the end to avoid these kind of scenarios:
`$query.name - $schedule.id - YYYYMMDDHHSS($schedule.timestamp)`

Reply:
Done in https://github.com/fleetdm/fleet/pull/12855/commits/1c34aafea1538799ae364a5071f5faed047ced94